auto_ria/auto_ria/spiders/ria_spider.py

```diff
@@ -9,25 +10,41 @@ class RiaSpider(scrapy.Spider):
     start_urls = []
     page_counter = 1
 
-    def __init__(self, url=None, *args, **kwargs):
+    def __init__(self, url=None, f=None, *args, **kwargs):
         super(RiaSpider, self).__init__(*args, **kwargs)
         self.search_url = url
+        self.input_file = f
+
+    def handle_params(self, url, filename):
```

Comment:
As I understand, `url` and `filename` are same variables as `self.search_url` and `self.input_file`.
In this case they should not be passed as parameters, because they are already available via `self`.